src/catalog/src/durable/impls/shadow.rs

```diff
@@ -88,7 +90,9 @@ where
         );
         let stash = stash?;
         let persist = persist?;
-        Ok(Box::new(ShadowCatalogState { stash, persist }))
+        let mut state = ShadowCatalogState { stash, persist };
```

Comment:
Would it make sense to wrap all this in a `let state = ShadowCatalogState::new(stash, persist);` function that builds the struct and calls fix_timestamps?

Reply:
Great idea! Just updated.